proposals/infra/INF-0005-Long-Vector-TestPlan.md

```diff
@@ -0,0 +1,273 @@
+# Test Plan
+
+All tests are to be included in the HLK test binary which ships with the OS.
+This test binary is only built in the OS repo and based off of the
+ExecutionTests source code in the DXC repo. There is a script in the WinTools
+repo which generates and annotates the HLK tests.
+
+There are three test categories we are concerned with:
+
+1. Implement DXIL Intrinsic tests:
+    - At the bottom of this document there is a table of all HLSL intrinsics and
+    their mapped DXIL OpCodes. We just need coverage for each DXIL OpCode. If
+    the DXIL OpCode column lists 'emulated' then this means that there are
+    multiple LLVM/DXIL Ops used to compute the intrinsic. We will need to do an
+    audit of them to confirm coverage of all vector ops.
+
+2. Implement LLVM Native operation tests:
+     - These are the test cases for the 'basic' HLSL operators listed in the
+     HLSL operators table below.
+     - We have added some test cases for these with the long vectors work, but
+     we will want to make sure we add HLK coverage for all of these.
+
+3. Standard loading and storing of long vectors
+     - Ensure we have some basic tests doing standard loading/storing of long
+     vectors across buffer types. __TODO: Add details about which types of
+     buffers to test and why.__
```

Comment:
> TODO: Add details about which types of buffers to test and why

This comment here to remind us that there's a TODO.  If we want to complete the PR without doing the TODO then we should get a follow-up issue filed.

Reply:
I'd prefer to fill in the details before completing this PR.